### PR TITLE
Refactor tests (again)

### PR DESCRIPTION
**Pull request by ds300**
Hello! This is a repeat of #1899 

The only change from last time is that we added `--runInBand` to the jest invocation to prevent the nondeterministic fork failure problem. Tests take about 3mins to run on CI now (before it was about 1:30). They still run in parallel locally, where the problem does not occur.

Not ideal, and we should continue debugging the jest/circle problem at some point. But I'd rather have slightly slower CI than lose these changes 😑

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 22ab7f5c8e34d60c4dbf1f7a9fb1776b267dd327

https://deploy-preview-1979--artsy-reaction.netlify.com

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "storybook": "node verify-node-version.js && concurrently --raw --kill-others 'yarn relay --watch' 'start-storybook --quiet -s ./public -p 9001'",
     "sync-schema": "curl https://raw.githubusercontent.com/artsy/metaphysics/master/_schema.graphql -o data/schema.graphql; yarn prettier --write --parser graphql data/schema.graphql",
     "sync-schema:localhost": "cd ../metaphysics && yarn dump-schema -- ../reaction/data",
-    "test": "node verify-node-version.js && jest",
+    "test": "node verify-node-version.js && jest --forceExit",
```
Comment (alloy) [inline package.json]: Is this still needed besides running in-band?

Comment (ds300) [reply package.json]: ah good catch, no, we should only have it in the circle config file.

**Comment by ds300** [comment]
@alloy should that be in the platform jira project?

**Comment by alloy** [comment]
Yeah let's do that 👍

**Comment by ds300** [comment]
Done https://artsyproduct.atlassian.net/browse/PLATFORM-1155

**Comment by damassi** [comment]
@ds300 - is that the right ticket link? Seems to point to Sidekiq.

**Comment by ds300** [comment]
@damassi Thanks for pointing that out. It was an off-by-one error. Weird.

**Comment by sweir27** [comment]
Cooool! It seems like this is good to go. Gonna merge and 🤞

**Comment by artsyit** [comment]
:rocket: PR was released in v12.1.16 :rocket:

**Pull request merged**